code/procs/jobprocs.dm

```diff
@@ -185,106 +190,96 @@
 			else if (istype(JOB, /datum/job/research/medical_doctor))
 				medical_staff += player
 
-			logTheThing("debug", null, null, "<b>I Said No/Jobs:</b> [player] took [JOB.name] from favorite selector")
-			player.mind.assigned_role = JOB.name
-			logTheThing("debug", player, null, "assigned job: [player.mind.assigned_role]")
-			unassigned -= player
-			JOB.assigned++
+			AssignJobToPlayer(player, JOB, "Favorite Job Picker", unassigned)
 
-	// Do this loop twice - once for med priority and once for low priority, because elsewise
-	// it was causing weird shit to happen where having something in low priority would
-	// sometimes cause you to get that instead of a higher prioritized job
-	for(var/datum/job/JOB in available_job_roles)
-		// If we've got everyone a job, then stop wasting cycles and get on with the show
-		if (unassigned.len == 0) break
-		// If there's no more slots for this job available, move onto the next one
-		if (JOB.limit > 0 && JOB.assigned >= JOB.limit) continue
-		// First, rebuild the lists of who wants to be this job
-		pick2 = FindOccupationCandidates(unassigned,JOB.name,2)
-		// Now loop through the candidates in order of priority, and elect them to the
-		// job position if possible - if at any point the job is filled, break the loops
-		for(var/mob/new_player/candidate in pick2)
-			if (istype(JOB, /datum/job/engineering/engineer))
-				engineering_staff += candidate
-			else if (istype(JOB, /datum/job/research/scientist))
-				research_staff += candidate
-			else if (istype(JOB, /datum/job/research/medical_doctor))
-				medical_staff += candidate
+	/////////////////////////////////
+	// ASSIGN MEDIUM PRIORITY JOBS //
+	/////////////////////////////////
 
-			if (JOB.assigned >= JOB.limit || unassigned.len == 0)
+	// Next go through each of the remaining players and see if we can assign them one of their medium priority jobs
+	for (var/mob/new_player/player in unassigned)
+		if (derelict_mode) // stop freaking out at the weird jobs
+			continue
+
+		// If they don't have any medium priority preferences, skip em
+		if (!player?.client?.preferences || player?.client?.preferences.jobs_med_priority.len == 0)
+			continue
+
+		var/list/medium_priority_preferences = player.client.preferences.jobs_med_priority
+		shuffle_list(medium_priority_preferences)
```

Comment:
If it's still desired to give a higher chance of getting a job that isn't considered low priority by the system, I could sort the player's job selection lists so that any other jobs they selected in that priority level that aren't "system low priority" are checked for availability first.